src/gpu/generic/sycl/rnn/ref_rnn.cpp

```diff
@@ -0,0 +1,791 @@
+
+    off.src_layer = get_outer_strides(src_layer_d);
+    off.src_iter = get_outer_strides(src_iter_d);
+
+    off.weights_layer = get_outer_strides(weights_layer_d);
+    off.weights_iter = get_outer_strides(weights_iter_d);
+
+    off.bias = get_outer_strides(bias_d);
+
+    off.dst_layer = get_outer_strides(dst_layer_d);
+    off.dst_iter = get_outer_strides(dst_iter_d);
+
+    if (!is_fwd) { return status::unimplemented; }
+
+    return status::success;
+}
+
+template <prop_kind_t aprop>
+inline status_t init_sycl(const rnn_utils::conf_t &rnn, const rnn_pd_t *rnn_pd,
+        rnn_offsets_t &off) {
+    return init_sycl(rnn_pd, rnn, rnn_pd->src_md(0), rnn_pd->src_md(1),
+            rnn_pd->weights_md(0), rnn_pd->weights_md(1), rnn_pd->weights_md(2),
+            rnn_pd->dst_md(0), rnn_pd->dst_md(1), rnn_pd->workspace_md(0), off);
+}
+
+template <>
+status_t _ref_rnn_common_t<prop_kind::forward>::pd_t::set_default_params() {
+    using namespace format_tag;
+    if (src_layer_md_.format_kind == format_kind::any)
+        CHECK(memory_desc_init_by_tag(src_layer_md_, tnc));
+    if (dst_layer_md_.format_kind == format_kind::any)
+        CHECK(memory_desc_init_by_tag(dst_layer_md_, tnc));
+
+    // Optional parameters
+    if ((!types::is_zero_md(&src_iter_md_))
+            && (src_iter_md_.format_kind == format_kind::any))
+        CHECK(memory_desc_init_by_tag(src_iter_md_, ldnc));
+    if ((!types::is_zero_md(&bias_md_))
+            && (bias_md_.format_kind == format_kind::any))
+        CHECK(memory_desc_init_by_tag(bias_md_, ldgo));
+    if ((!types::is_zero_md(&dst_iter_md_))
+            && (dst_iter_md_.format_kind == format_kind::any))
+        CHECK(memory_desc_init_by_tag(dst_iter_md_, ldnc));
+
+    return status::success;
+}
+
+template <prop_kind_t aprop>
+status_t _ref_rnn_common_t<aprop>::pd_t::init(impl::engine_t *engine) {
+    using namespace prop_kind;
+    using namespace utils;
+    using namespace rnn_utils;
+    using namespace format_tag;
+
+    assert(engine->kind() == engine_kind::gpu);
+
+    const alg_kind_t cell_kind = this->desc()->cell_kind;
+
+    data_type_t src_layer_dt = this->desc()->src_layer_desc.data_type;
+    data_type_t weights_iter_dt = this->desc()->weights_iter_desc.data_type;
+    data_type_t weights_layer_dt = this->desc()->weights_layer_desc.data_type;
+    data_type_t bias_dt = this->desc()->bias_desc.data_type;
+
+    acc_data_t = data_type::f32;
+
+    src_type = src_layer_dt;
+    weights_type = weights_layer_dt;
+
+    VDISPATCH_RNN(
+            one_of(cell_kind, alg_kind::vanilla_rnn), VERBOSE_BAD_ALGORITHM);
+    VDISPATCH_RNN(IMPLICATION(aprop == prop_kind::forward,
+                          one_of(this->desc()->prop_kind, forward_training,
+                                  forward_inference)),
+            VERBOSE_BAD_PROPKIND);
+    VDISPATCH_RNN(
+            IMPLICATION(src_type == data_type::bf16, bias_dt == data_type::f32),
+            VERBOSE_UNSUPPORTED_DT);
```

Comment:
This check seems unnecessary given that bias doesn't seem to be supported due to the following check:
```
 VDISPATCH_RNN(this->with_bias(), VERBOSE_UNSUPPORTED_BIAS_CFG);
```

Reply:
Unneeded check removed